slidechain/export.go

```diff
@@ -58,34 +58,33 @@ const (
+	              #                           value, json, {exporter}       
+	get get get   #  {exporter}, json, value                                  
 	x'%x' output  #                                                  {O,...}
 `
 
-	// TODO(debnil): Modify this contract to produce a zeroval from the value it contains,
-	// so the caller does not have to do a new `nonce`.
 	exportContract2Fmt = `
-	                     #  con stack                          arg stack                 log
-	                     #  ---------                          ---------                 ---
-	                     #  {exporter}, value, json            selector                                    
-	get                  #  {exporter}, value, json, selector                                              
-	jumpif:$doretire     #                                                                                 
-	                     #  {exporter}, value, json                                                        
-	"" put               #  {exporter}, value, json            ""                                          
-	drop                 #  {exporter}, value                                                              
-	put put 1 put        #                                     "", value, {exporter}, 1                    
-	x'%x' contract call  #                                                               {'L',...}{'O',...}
-	jump:$checksig       #                                                                                   
-	                     #                                                                                   
-	$doretire            #                                                                                   
-	                     #  {exporter}, value, json                                                          
-	put put drop         #                                     json, value                                   
-	x'%x' contract call  #                                                                                   
-	                     #                                                                                                        
-	$checksig            #                                                                                   
-	[%s] yield           #                                     sigchecker                                    
-                                                                                   
+	                     #  con stack                                   arg stack                 log
+	                     #  ---------                                   ---------                 ---
+	                     #  {exporter}, json, val                       selector                                      
+	0 split 3 bury swap  #  zeroval, {exporter}, value, json            selector                                      
```

Comment:
TxVM style nit[*] - prefer the assembler macro `splitzero` over `0 split`.

[*] - ATTINWDTS[**]
[**] - Acknowledging that there is no well-defined TxVM style

Reply:
Those empty brackets were supposed to contain a double star...